In [1]:
import numpy as np

### Урок 5. Проверка статистических гипотез. P-значения. Доверительные интервалы. A/B-тестирование

#### Задание №1
1) Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением, равным 16.Найти доверительный интервал для оценки математического ожидания с надежностью 0.95,если выборочная средняя M = 80, а объем выборки n = 256.

In [2]:
# Доверительный интервал
di1 = 80 + 1.96 * 16 / 256**0.5
di1

81.96

In [3]:
di2 = 80 - 1.96 * 16 / 256**0.5
di2

78.04

Доверительный интервал будет от 78.04 до 81.96 

#### Задание №2
2) В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные: 6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1 Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95.

In [4]:
a = [6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1]
len(a)

10

In [5]:
# Найдем несмещенное среднеквадратичное откланение
std_a = np.std(a, ddof=1)
std_a

0.4508017549014448

In [6]:
# Найдем среднее
mean_a = np.mean(a)
mean_a

6.590000000000001

In [7]:
#Найдем доверительный интервал
di1 = 6.59 + 2.262 * std_a / len(a)**0.5
di1 

6.912461744087574

In [8]:
di2= 6.59 - 2.262 * std_a / len(a)**0.5
di2

6.2675382559124255

Доверительный интервал будет от 6.26 до 6.91

#### Задание №3
Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметроказался равным 17.5 мм, а дисперсия известна и равна 4 кв.мм.

In [9]:
# H0: X = m
# H1: X > m
Zh = (17.5 - 17 ) / (4**0.5 / 100**0.5)
Zh

2.5

Нулевая гипотеза не верна, так как Z критерий для 5% равен 1.645 а для выборки он больше и равен 2.5

4) Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек.
Вес каждой пачки составляет:
202, 203, 199, 197, 195, 201, 200, 204, 194, 190. Известно,
что их веса распределены нормально.
Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?

In [10]:
# H0: X = m
# H1: X /= m
b = [202, 203, 199, 197, 195, 201, 200, 204, 194, 190]
len(b)

10

In [11]:
# Найдем несмещенное среднеквадратичное откланение
std_b = np.std(b, ddof=1)
std_b

4.453463071962462

In [12]:
# Найдем среднее
mean_b = np.mean(b)
mean_b

198.5

In [13]:
# Критерий Стьюденса
tb = (200 - mean_b ) / (std_b / len(b)**0.5)
tb

1.0651074037450896

Нулевая гипотеза верна, так как t-критерий для 0.05% равен 3.6897 а для выборки он равен 1.06